dev/AppNotifications/WICUtility.cpp

```diff
@@ -0,0 +1,166 @@
+    winrt::com_ptr<IWICImagingFactory> const& pWICImagingFactory,
+    winrt::com_ptr<IWICBitmapSource> const& pWICBitmapSource,
+    WICPixelFormatGUID guidPixelFormatSource,
+    WICBitmapDitherType bitmapDitherType)
+{
+    winrt::com_ptr<IWICFormatConverter> spWICFormatConverter;
+
+    THROW_IF_FAILED(pWICImagingFactory->CreateFormatConverter(spWICFormatConverter.put()));
+
+    THROW_IF_FAILED(spWICFormatConverter->Initialize(pWICBitmapSource.get(), guidPixelFormatSource, bitmapDitherType, nullptr, 0.f, WICBitmapPaletteTypeCustom));
+
+    // Store the converted bitmap as ppToRenderBitmapSource
+    winrt::com_ptr<IWICBitmapSource> ppWICBitmapSource;
+    THROW_IF_FAILED(spWICFormatConverter->QueryInterface(_uuidof(IWICBitmapSource), ppWICBitmapSource.put_void()));
+
+    return ppWICBitmapSource;
+}
+
+void AddFrameToWICBitmap(
+    winrt::com_ptr<IWICImagingFactory> const& pWICImagingFactory,
+    winrt::com_ptr<IWICBitmapEncoder> const& pWICBitmapEncoder,
+    winrt::com_ptr<IWICBitmapSource> const& pWICBitmapSource,
+    BITMAP_VERSION bmpv)
+{
+    winrt::com_ptr<IWICBitmapFrameEncode> spWICFrameEncoder;
+    winrt::com_ptr<IPropertyBag2> spWICEncoderOptions;
+
+    THROW_IF_FAILED(pWICBitmapEncoder->CreateNewFrame(spWICFrameEncoder.put(), spWICEncoderOptions.put()));
+
+    GUID containerGuid;
+    THROW_IF_FAILED(pWICBitmapEncoder->GetContainerFormat(&containerGuid));
+
+    if ((containerGuid == GUID_ContainerFormatBmp) && (bmpv == BMPV_5))
+    {
+        // Write the encoder option to the property bag instance.
+        VARIANT varValue{};
+
+        varValue.vt = VT_BOOL;
+        varValue.boolVal = VARIANT_TRUE;
+
+        // Options to enable the v5 header support for 32bppBGRA.
+        PROPBAG2 v5HeaderOption{};
+
+        std::wstring str = L"EnableV5Header32bppBGRA";
+        v5HeaderOption.pstrName = (LPOLESTR)str.c_str();
+
+        THROW_IF_FAILED(spWICEncoderOptions->Write(1, &v5HeaderOption, &varValue));
+    }
+
+    THROW_IF_FAILED(spWICFrameEncoder->Initialize(spWICEncoderOptions.get()));
+
+    UINT uWidth, uHeight;
+    THROW_IF_FAILED(pWICBitmapSource->GetSize(&uWidth, &uHeight));
+
+    THROW_IF_FAILED(spWICFrameEncoder->SetSize(uWidth, uHeight));
+
+    winrt::com_ptr<IWICBitmapSource> spBitmapSourceConverted;
+    spBitmapSourceConverted = ConvertWICBitmapPixelFormat(pWICImagingFactory, pWICBitmapSource, GUID_WICPixelFormat32bppBGRA, WICBitmapDitherTypeNone);
+
+    WICRect rect { 0, 0, static_cast<INT>(uWidth), static_cast<INT>(uHeight) };
+
+    // Write the image data and commit
+    THROW_IF_FAILED(spWICFrameEncoder->WriteSource(spBitmapSourceConverted.get(), &rect));
+
+    THROW_IF_FAILED(spWICFrameEncoder->Commit());
+}
+
+winrt::com_ptr<IStream> GetStreamOfWICBitmapSource(
+    winrt::com_ptr<IWICImagingFactory> const& pWICImagingFactory,
+    winrt::com_ptr<IWICBitmapSource> const& pWICBitmapSource,
+    _In_ REFGUID guidContainerFormat,
+    _In_ BITMAP_VERSION bmpv)
```

Comment:
Please define more descriptive parameter names.

Reply:
Will be followed up with a newer update.